src/components/NamespaceDropdown.tsx

```diff
@@ -55,8 +58,25 @@ export class NamespaceDropdown extends React.PureComponent<NamespaceListType, {}
 
   componentDidMount() {
     this.props.refresh();
+    this.syncNamespacesURLParam();
   }
 
+  componentDidUpdate(prevProps: NamespaceListType) {
+    if (prevProps.activeNamespaces !== this.props.activeNamespaces) {
+      HistoryManager.setParam(URLParams.NAMESPACES, this.props.activeNamespaces.map(item => item.name).join(','));
+    }
+  }
+
+  syncNamespacesURLParam = () => {
+    const namespaces = (HistoryManager.getParam(URLParams.NAMESPACES) || '').split(',').filter(Boolean);
+    if (namespaces.length > 0 && _.difference(namespaces, this.props.activeNamespaces.map(item => item.name))) {
+      // We must change the props of namespaces
+      const items: Namespace[] = [];
+      namespaces.map((ns: string) => items.push({ name: ns }));
```

Comment:
this should be one-liner `const items = namespaces.map(ns => { name: ns });`

Reply:
Yes ...I don't know what IO was thinking ^^ good review

Reply:
Solution : const items = namespaces.map(ns => ({ name: ns } as Namespace))